### PR TITLE
Replace cpp keywords used as symbol names

### PR DESCRIPTION
**Pull request by mmaka1**
Current output of FW API documentation is corrupted in some places when *Breathe* (a bridge between doxygen output and Sphinx) is importing symbols into *c* domain:

```c
enum comp_copy_type {
	COMP_COPY_NORMAL = 0,	/**< Normal */
	COMP_COPY_BLOCKING,	/**< Blocking */
```
gives (removed formatting, see the missing name of item init'ed with value):

>Values:
> 0
> Normal.
> COMP_COPY_BLOCKING

```c
struct comp_ops {
...
	int (*params)(struct comp_dev *dev,
		      struct sof_ipc_stream_params *params);
	
```
gives (see missing ` (*` before `params`):

> intparams)(struct comp_dev *dev, struct sof_ipc_stream_params *params)

When Breathe is switched to cpp domain (found that Zephyr for instance is using cpp domain for api documentation as well), the output looks much better:
>Values:
> COMP_COPY_NORMAL = 0
> Normal.
> COMP_COPY_BLOCKING

> int (*params)(struct comp_dev *dev, struct sof_ipc_stream_params *params)

But when running in cpp domain, Sphinx outputs errors for all cpp keywords used as symbol names and generates wrong output for them. There are two used in a the documented headers: `new` and `private`. This set of patches is going to replace them with some neutral symbol names.

### CODE REVIEW

**Comment by jajanusz** [comment]
Is there no switch to make doxygen *smarter*, so it treats it like C code not C++ ? I'd prefer to not change code just because of doxygen.

**Comment by mmaka1** [comment]
@jajanusz This is not doxygen (configuration already optimized for processing headers as *C* code), neither Breathe, it is Sphinx, the last one in the processing chain, see my original description:

> when running in cpp domain, Sphinx outputs errors

File: src/drivers/generic/dummy-dma.c [hunk 1]
```diff
@@ -458,7 +458,7 @@ static int dummy_dma_probe(struct dma *dma)
 		dma->chan[i].dma = dma;
 		dma->chan[i].index = i;
 		dma->chan[i].status = COMP_STATE_INIT;
-		dma->chan[i].private = &chanp[i];
+		dma->chan[i].priv_data = &chanp[i];
```
Comment (paulstelian97) [inline src/drivers/generic/dummy-dma.c]: Can you replace it with `dma_chan_set_data(&dma->chan[i], &chanp[i]);`? In a separate patch. Thank you for showing that I am accessing this directly which isn't the best thing to do.

Comment (paulstelian97) [reply src/drivers/generic/dummy-dma.c]: Thanks.

**Pull request merged**